llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7759,7 +7535,10 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
   }
 
   for (auto &P : VPlans) {
-    for (ElementCount VF : P->vectorFactors()) {
+    SmallVector<ElementCount, 1> VFs(P->vectorFactors());
+    auto RUs = ::calculateRegisterUsage(*P, VFs, TTI);
+    for (unsigned I = 0; I < VFs.size(); I++) {
+      auto VF = VFs[I];
```

Comment:
Nit, can you use a zip here?
```suggestion
    for (auto [VF, RU] : zip(VFs, RUs)) {
```